### PR TITLE
Add projection of step metrics on skeleton planes, decrease connections load and move console connections to yarp application

### PR DESCRIPTION
**Pull request by mfussi66**
This PR restores the original plan of projecting the step metrics (such as length, width) along the planes defined by the human body (sagittal, coronal, transverse).
This is now possible after fixing the getters for the planes, see the commit: https://github.com/robotology/assistive-rehab/commit/75aa76954fe73309d7fd32f026d08620f7451a8 The getters included a gaze2base transformation that was already embedded in the skeleton keypoints postions.
This can be seen in the following piece of code in https://github.com/robotology/assistive-rehab/blob/master/lib/src/skeleton.cpp:
```diff
void Skeleton::update()
{
    Vector p(4,1.0);
    for (auto &k:keypoints)
    {
        if (k->isUpdated())
        {
            auto v=k->getPoint();
            auto px=k->getPixel();
            p[0]=v[0];
            p[1]=v[1];
            p[2]=v[2];
+            k->setPoint((T*p).subVector(0,2),px);
        }
    }

+    update_planes();
}

```
The Keypoints are transformed through the T matrix, then the `update_planes()` function is called. The function updates coronal, sagittal, and transverse planes with the newly evaluated keypoints. Using the getter in the form of:
```
Vector Skeleton::getCoronal() const
{
    return (T.submatrix(0,2,0,2)*coronal);
}
```

would cause the transformation T to be applied twice.

**Note:** The transformation T is only used to transform the Keypoints.

----

Some other changes:

For debugging purposes, step length and width are now plotted thanks to a dedicated port `/motionAnalyzer/scopeRaw`.

Fixes made to lower the load on the machines:
- decreased the logging logging frequency of some prints of obstacleDetector (the ones who are not crucial to the robot safety)
- Added proper compression on OpenPose -> retriever connection
- Moved the connections of the docker-compose-console nodes from docker to the simple yarp manager application: in this way we don't create 50 containers for just making yarp connections

cc @ste93 @elandini84 @randaz81

### CODE REVIEW

**Comment by pattacini** [comment]
Seems there's an ongoing dev on this.
Put in draft.

**Comment by mfussi66** [comment]
> Seems there's an ongoing dev on this. Put in draft.

Yes sorry about that, we noticed that the modifications in the latest 3 commits were necessary for the demo. I'll update the PR description.

File: modules/skeletonRetriever/src/main.cpp [hunk 1]
```diff
@@ -953,6 +953,9 @@ class Retriever : public RFModule
         update_gaze_frame();
         rootFrame=navFrame*gazeFrame;
 
+        // yDebug() << "NAV\n" << navFrame.toString();
+        // yDebug() << "GAZE\n" << gazeFrame.toString();
+
```
Comment (pattacini) [inline modules/skeletonRetriever/src/main.cpp:958]: Please remove these lines.

File: modules/speechInteraction/modules/googleSpeech/main.cpp [hunk 2]
```diff
@@ -273,10 +273,6 @@ class Processing : public yarp::os::TypedReaderCallback<yarp::sig::Sound>
                 b.addString(alternative.transcript());
             }
         }
-        
-
-
-
```
Comment (pattacini) [inline modules/speechInteraction/modules/googleSpeech/main.cpp:279]: Please remove these empty lines.

File: modules/speechInteraction/modules/googleSpeech/main.cpp [hunk 3]
```diff
@@ -410,6 +406,163 @@ class Module : public yarp::os::RFModule, public googleSpeech_IDL
 };
 
 /********************************************************/
+<<<<<<< Updated upstream
```
Comment (pattacini) [inline modules/speechInteraction/modules/googleSpeech/main.cpp:409]: It seems there have been problems merging this file.

File: modules/speechInteraction/modules/googleSpeech/main.cpp [hunk 4]
```diff
@@ -410,6 +406,163 @@ class Module : public yarp::os::RFModule, public googleSpeech_IDL
...
+    /********************************************************/
+    bool updateModule()
+    {
+        return !closing;
+    }
+};
+
+
+/********************************************************/
+>>>>>>> Stashed changes
```
Comment (pattacini) [inline modules/speechInteraction/modules/googleSpeech/main.cpp:565]: Same as above.

File: modules/motionAnalyzer/src/Manager.cpp [hunk 5]
```diff
@@ -873,15 +874,15 @@ Property Manager::getState()
 /********************************************************/
 bool Manager::isStanding()
 {
-    yInfo()<<"shoulder height speed"<<shoulder_center_height_vel;
+    //yInfo()<<"shoulder height speed"<<shoulder_center_height_vel;
```
Comment (pattacini) [inline modules/motionAnalyzer/src/Manager.cpp:877]: Uncomment or remove the line.

File: modules/motionAnalyzer/src/Manager.cpp [hunk 6]
```diff
@@ -873,15 +874,15 @@ Property Manager::getState()
...
     standing=(shoulder_center_height_vel>standing_thresh);
     return standing;
 }
 
 /********************************************************/
 bool Manager::isSitting()
 {
-    yInfo()<<"shoulder height speed"<<shoulder_center_height_vel;
+    //yInfo()<<"shoulder height speed"<<shoulder_center_height_vel;
```
Comment (pattacini) [inline modules/motionAnalyzer/src/Manager.cpp:885]: As above.

File: modules/motionAnalyzer/src/Processor.cpp [hunk 7]
```diff
@@ -315,9 +346,13 @@ void Step_Processor::estimateSpatialParams(const double &dist,const double &widt
...
     }
 
+    steplen_raw = output1[0];
+    stepwidth_raw = output2[0];
     steplen=0.0;
     stepwidth=0.0;
-    if ( (Time::now()-tlast)<=time_window )
+    if ((Time::now()-tlast)<=time_window
+         //steplen_raw > step_thresh
+         )
```
Comment (pattacini) [inline modules/motionAnalyzer/src/Processor.cpp:355]: Clean up the commented lines.

File: lib/src/skeleton.cpp [hunk 8]
```diff
@@ -339,17 +339,17 @@ bool Skeleton::setTransverse(const Vector &transverse)
...
 Vector Skeleton::getSagittal() const
 {
-    return (T.submatrix(0,2,0,2)*sagittal);
+    return (sagittal);
 }
 
 Vector Skeleton::getTransverse() const
 {
-    return (T.submatrix(0,2,0,2)*transverse);
+    return (transverse);
```
Comment (pattacini) [inline lib/src/skeleton.cpp:352]: Please remove the unnecessary parentheses.

**Review by pattacini — CHANGES_REQUESTED** [review]
Hi @mfussi66 

A few simple points to address.

Regarding

>This PR restores the original plan of projecting the step metrics (such as length, width) along the planes defined by the human body (sagittal, coronal, transverse).
This is now possible after fixing the getters for the planes, see the commit: https://github.com/robotology/assistive-rehab/commit/75aa76954fe73309d7fd32f026d08620f7451a8 The getters included a gaze2base transformation that was already embedded in the skeleton keypoints positions.

would you please trace back the parts in the code where the transformation is done? This is for the record.

File: lib/src/skeleton.cpp (hunk 8 above)
Comment (mfussi66) [reply lib/src/skeleton.cpp]: Done ✅

File: modules/motionAnalyzer/src/Manager.cpp (hunk 6 above)
Comment (mfussi66) [reply modules/motionAnalyzer/src/Manager.cpp]: Done ✅

File: modules/motionAnalyzer/src/Manager.cpp (hunk 5 above)
Comment (mfussi66) [reply modules/motionAnalyzer/src/Manager.cpp]: Done ✅

File: modules/speechInteraction/modules/googleSpeech/main.cpp (hunk 4 above)
Comment (mfussi66) [reply modules/speechInteraction/modules/googleSpeech/main.cpp:563]: Done ✅

File: modules/speechInteraction/modules/googleSpeech/main.cpp (hunk 2 above)
Comment (mfussi66) [reply modules/speechInteraction/modules/googleSpeech/main.cpp:279]: Done ✅

File: modules/speechInteraction/modules/googleSpeech/main.cpp (hunk 3 above)
Comment (mfussi66) [reply modules/speechInteraction/modules/googleSpeech/main.cpp]: I must have missed the line when merging locally, I will double check if the speech modules compile fine.

File: modules/skeletonRetriever/src/main.cpp (hunk 1 above)
Comment (mfussi66) [reply modules/skeletonRetriever/src/main.cpp]: Done ✅

File: modules/speechInteraction/modules/googleSpeech/main.cpp (hunk 4 above)
Comment (pattacini) [reply modules/speechInteraction/modules/googleSpeech/main.cpp:563]: There's still the `>>>>>>> Stashed changes` to fix.

File: modules/motionAnalyzer/src/Processor.cpp (hunk 7 above)
Comment (mfussi66) [reply modules/motionAnalyzer/src/Processor.cpp]: Done

File: modules/speechInteraction/modules/googleSpeech/main.cpp (hunk 4 above)
Comment (mfussi66) [reply modules/speechInteraction/modules/googleSpeech/main.cpp]: Done.

File: modules/speechInteraction/modules/googleSpeech/main.cpp (hunk 3 above)
Comment (mfussi66) [reply modules/speechInteraction/modules/googleSpeech/main.cpp]: Speech modules have been reverted to master, they need to be kept untouched in this pr.

**Comment by mfussi66** [comment]
> would you please trace back the parts in the code where the transformation is done? This is for the record.
@pattacini The transformation matrix is computed by the `skeletonRetriever`, which computes the transformation from gaze to base, by taking info from the gaze controller and the navigation controlled. 
Then the `update()` function of the skeleton updates the keypoints position by transforming the position in the gaze frame to the base frame. 
The `update()` also is tasked with updating the skeleton planes.

File: modules/speechInteraction/modules/googleSpeech/main.cpp [hunk 9]
```diff
@@ -273,11 +273,6 @@ class Processing : public yarp::os::TypedReaderCallback<yarp::sig::Sound>
                 b.addString(alternative.transcript());
             }
         }
-        
-
-
-
-
         return b;
     }
```
Comment (pattacini) [inline modules/speechInteraction/modules/googleSpeech/main.cpp:278]: To avoid conflicts with the upcoming PR's, better off resetting this file outright.

**Review by pattacini — CHANGES_REQUESTED** [review]
Just a final minor change is needed.

File: modules/speechInteraction/modules/googleSpeech/main.cpp (hunk 9 above)
Comment (mfussi66) [reply modules/speechInteraction/modules/googleSpeech/main.cpp]: Resetted!

**Review by pattacini — APPROVED** [review]

**Pull request merged**